src/evm/protocol/uniswap_v3/state.rs

```diff
@@ -221,6 +227,104 @@ impl UniswapV3State {
             sqrt_price_next
         }
     }
+
+    /// Converts a target price to sqrt_price_x96 format with fee adjustment
+    ///
+    /// # Arguments
+    /// * `token_in` - The token being sold
+    /// * `token_out` - The token being bought
+    /// * `target_price` - The target price as token_out/token_in (tycho convention)
+    ///
+    /// # Returns
+    /// The sqrt price limit in Q96 format
+    fn get_sqrt_price_limit(
+        &self,
+        token_in: &Bytes,
+        token_out: &Bytes,
+        target_price: &Price,
```

Comment:
I think price can implement Copy trait. you don't need to pass by reference here